CSharp/DevVmPowershell/Helpers/DisclaimerAcceptanceHelper.cs

```diff
@@ -0,0 +1,295 @@
+			query.Fields = FieldValue.AllFields;
+			try
+			{
+				kCura.Relativity.Client.DTOs.QueryResultSet<Workspace> resultSet =
+					RsapiClient.Repositories.Workspace.Query(query, 0);
+				if (resultSet.Success && resultSet.Results.Count > 0)
+				{
+					workspaceId = resultSet.Results.First().Artifact.ArtifactID;
+				}
+				else
+				{
+					throw new Exception($"Unable to find workspace with the name: {workspaceName}");
+				}
+			}
+			catch (Exception ex)
+			{
+				throw new Exception($"Error finding workspace with name: {workspaceName}", ex);
+			}
+
+			return workspaceId;
+		}
+
+		private int GetDisclaimerSolutionConfigurationObjectTypeId(int workspaceId)
+		{
+			int objectTypeId;
+			RsapiClient.APIOptions.WorkspaceID = workspaceId;
+			try
+			{
+				ResultSet<ObjectType> results = RsapiClient.Repositories.ObjectType.Read(
+					new ObjectType(new Guid(Constants.DisclaimerAcceptance.ObjectGuids.DisclaimerSolutionConfiguration))
+					{
+						Fields = FieldValue.AllFields
+					});
+				if (!results.Success)
+				{
+					throw new Exception(
+						$"Reading {Constants.DisclaimerAcceptance.ObjectNames.DisclaimerSolutionConfiguration} ObjectType was not successful");
+				}
+
+				objectTypeId = results.Results.First().Artifact.DescriptorArtifactTypeID.Value;
+			}
+			catch (Exception ex)
+			{
+				throw new Exception(
+					$"Error Reading the {Constants.DisclaimerAcceptance.ObjectNames.DisclaimerSolutionConfiguration} ObjectType", ex);
+			}
+
+			return objectTypeId;
+		}
+
+		private int GetDisclaimerLayoutId()
+		{
+			int layoutId;
+			TextCondition queryCondition =
+				new TextCondition(LayoutFieldNames.Name, TextConditionEnum.EqualTo, Constants.DisclaimerAcceptance.LayoutNames.DisclaimerLayout);
+			Query<kCura.Relativity.Client.DTOs.Layout> layoutQuery = new Query<kCura.Relativity.Client.DTOs.Layout>();
+			layoutQuery.Condition = queryCondition;
+			layoutQuery.Fields = FieldValue.AllFields;
+			ResultSet<kCura.Relativity.Client.DTOs.Layout> layoutQueryResultSet =
+				new ResultSet<kCura.Relativity.Client.DTOs.Layout>();
+			try
+			{
+				layoutQueryResultSet = RsapiClient.Repositories.Layout.Query(layoutQuery);
+				layoutId = layoutQueryResultSet.Results.First().Artifact.ArtifactID;
+			}
+			catch (Exception ex)
+			{
+				throw new Exception($"Error Reading {Constants.DisclaimerAcceptance.LayoutNames.DisclaimerSolutionConfigurationLayout} Id", ex);
+			}
+
+			return layoutId;
+		}
+
+		private int GetDisclaimerSolutionConfigurationLayoutId()
+		{
+			int layoutId;
+			TextCondition queryCondition =
+				new TextCondition(LayoutFieldNames.Name, TextConditionEnum.EqualTo, Constants.DisclaimerAcceptance.LayoutNames.DisclaimerSolutionConfigurationLayout);
+			Query<kCura.Relativity.Client.DTOs.Layout> layoutQuery = new Query<kCura.Relativity.Client.DTOs.Layout>();
+			layoutQuery.Condition = queryCondition;
+			layoutQuery.Fields = FieldValue.AllFields;
```

Comment:
if you are only getting the ArtifactId, you can do FieldValue.NoFields which is much more efficient than doing all fields.